storm-core/src/jvm/org/apache/storm/topology/BaseConfigurationDeclarer.java

```diff
@@ -56,7 +56,11 @@ public T setNumTasks(Number val) {
 
     @Override
     public T setMemoryLoad(Number onHeap) {
-        return setMemoryLoad(onHeap, Utils.getDouble(conf.get(Config.TOPOLOGY_COMPONENT_RESOURCES_OFFHEAP_MEMORY_MB)));
+        if (onHeap != null) {
+            onHeap = onHeap.doubleValue();
+            return addConfiguration(Config.TOPOLOGY_COMPONENT_RESOURCES_ONHEAP_MEMORY_MB, onHeap);
+        }
+        return null;
```

Comment:
Minor: This can be replaced with `setMemoryLoad(onHeap, null)` but I'm OK to split two methods completely.